scripts/recovery_sync

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+#
+# This script will sync host keys, authorized public keys, and networking
+# configuration from the running sytem to the recovery environment. It should
+# be run whenever any of these things change, to ensure that the recovery
+# environment remains accessible in the event of a serious system problem.
+#
+
+set -euxo pipefail
+
+function die() {
+	echo "$1" >&2
+	exit 1
+}
+
+[[ $# -eq 1 ]] || die "Illegal number of parameters"
+target="$1"
+workdir=$(mktemp -d)
+
+cd "$workdir"
+chown root .
+chmod 0755 .
+gunzip <"$target" | cpio -idmv
+
+rsync -a /run/systemd/network/ etc/systemd/network
+mkdir -p etc/dropbear
+for fmt in $(find /etc/ssh -name \*key | sed 's/.*host_\(.*\)_key/\1/'); do
+	# Unfortunately, dropbear doesn't support ed25519 keys.
+	[[ "$fmt" == "ed25519" ]] && continue
+	./bin/dropbearconvert openssh dropbear "/etc/ssh/ssh_host_${fmt}_key" "etc/dropbear/dropbear_${fmt}_host_key"
+done
```

Comment:
What does this do?

Reply:
This converts the system host keys to a format that dropbear can read. I'll add a comment.